docs/content/5.integrations/2.guide/2.adding-headers-to-request.md

```diff
@@ -0,0 +1,284 @@
+# Pass Custom Headers in Requests
+
+In this tutorial, we'll demonstrate how to set up an interceptor to pass headers when making requests using the axios client inside the Vue Storefront SDK. This is useful when you need to append custom headers or cookies to your API calls.
+
+After passing the headers to the axios client, we'll also show you how to pass them to external services using the API Client.
+
+::danger
+Do not use createSharedComposable if creating a shared composable that uses the interceptor. This will cause the interceptor to only be created once, which will result in the headers being shared across instances.
+::
+
+## Prerequisites:
+Ensure you have the Vue Storefront SDK set up in your project.
+
+## Notes
+::info
+This guide uses the integrations boilerplate with the nuxt option. The same basic principles apply to other frameworks, but the implementation may vary.
+::
+
+All SDK modules export a client instance of axios. This is the axios instance that is used to make requests. You can access this instance by importing the `client` from the SDK.
+
+## Steps
```

Comment:
Would be great to have some more in-depth explanations of what is happening here and why. Since implementations can vary it's important for end-user to understand WHY these things should happen even more than WHAT should happen. Then they can adjust it to their projects.